implementations/rust/ockam/ockam_app/src/app/state/mod.rs

```diff
@@ -331,3 +367,69 @@ fn load_model_state(
         }
     })
 }
+
+pub type EventName = String;
+type IsProcessing = AtomicBool;
+
+struct EventManager {
+    events: HashMap<EventName, IsProcessing>,
```

Comment:
We can simplify using a Vec<>

Reply:
You mean storing a tuple in the vector for each item? `Vec<(EventName, IsProcessing)>`